src/go/rpk/pkg/cli/generate/prometheus.go

```diff
@@ -42,60 +47,91 @@ type prometheusTLS struct {
 	KeyFile  string `yaml:"key_file,omitempty"`
 }
 
+///////////
+// Flags //
+///////////
+
 type tlsConfig struct {
 	TLS       config.TLS
 	enableTLS bool
 }
 
+type prometheusOptions struct {
+	intMetrics bool
+	jobName    string
+	labels     []string
+	nodeAddrs  []string
+	seedAddr   string
+}
+
 func newPrometheusConfigCmd(fs afero.Fs, p *config.Params) *cobra.Command {
 	var (
-		intMetrics bool
-		jobName    string
-		nodeAddrs  []string
-		seedAddr   string
-		tlsCfg     tlsConfig
+		opts   prometheusOptions
+		tlsCfg tlsConfig
 	)
 	cmd := &cobra.Command{
 		Use:   "prometheus-config",
-		Short: "Generate the Prometheus configuration to scrape redpanda nodes",
-		Long: `
-Generate the Prometheus configuration to scrape redpanda nodes. This command's
-output should be added to the 'scrape_configs' array in your Prometheus
-instance's YAML config file.
-
-If --seed-addr is passed, it will be used to discover the rest of the cluster
-hosts via Redpanda's Kafka API. If --node-addrs is passed, they will be used
-directly. Otherwise, 'rpk generate prometheus-conf' will read the redpanda
-config file and use the node IP configured there. --config may be passed to
-specify an arbitrary config file.
-
-If the node you want to scrape uses TLS, you can provide the TLS flags:
---tls-key, --tls-cert, and --tls-truststore and rpk will generate the required
-tls_config section in the scrape configuration.`,
+		Short: "Generate the Prometheus configuration to scrape Redpanda nodes",
+		Long: `Generate the Prometheus configuration to scrape Redpanda nodes. 
+
+The output of this command should be included in the 'scrape_configs' array 
+within the YAML configuration file of your Prometheus instance.
+
+There are different options you can use when generating the configuration:
+
+ - If you provide the --seed-addr flag, the command will use the address to 
+   discover the rest of the cluster hosts using Redpanda's Kafka API.
+ - If you provide the --node-addrs flag, the command will directly use the 
+   provided addresses.
+ - If neither --seed-addr nor --node-addrs are passed, the command will read the 
+   redpanda config file and use the node IP configured there.
+
+If the node you want to scrape uses TLS, you can provide the TLS flags 
+(--tls-key, --tls-cert, and --tls-truststore). The command will generate the 
+required tls_config section in the scrape configuration.
+
+Additionally, you can specify common labels for the target in the static-config 
+section using the --labels flag. You can also specify which metric you want the 
+label to target. For example:
+
+  --job-name test --labels "ext|group:one,int|group:two"
```

Comment:
This help text doesn't say what ext or int are, and I don't understand how they map to the yaml in the comment just below?

Reply:
Changed, moved the help text to a variable.